### PR TITLE
Added basic auth traffic policy action

### PR DESCRIPTION
**Pull request by ForgoneReality**

### CODE REVIEW

File: traffic-policy/actions/jwt-validation/examples/basic-example.mdx [hunk 1]
```diff
@@ -3,7 +3,7 @@ import ConfigExample from "/src/components/ConfigExample.tsx";
 ### Basic Example
 
 The following [Traffic Policy](/docs/http/traffic-policy/)
-configuration is an example configuration of the `jwt-validation` action for a
+configuration is an example configuration of the `jwt-validation` action. For a
```
Comment (nijikokun) [inline traffic-policy/actions/jwt-validation/examples/basic-example.mdx:6]: This change should be outside of this PR.

**Review by nijikokun — CHANGES_REQUESTED** [review]

File: traffic-policy/actions/basic-auth/index.mdx [hunk 2]
```diff
@@ -0,0 +1,6 @@
+## Overview
+
+The **Basic Auth** Traffic Policy action enables you to enforce HTTP Basic
+Authentication in front of your endpoints. Requests with a valid username
```
[the remark this answers is not in the document]
Comment (ForgoneReality) [reply traffic-policy/actions/basic-auth/index.mdx:4]: Do you want to cut the entire thing into that one-liner, or just change the first sentence?

File: traffic-policy/actions/basic-auth/behavior.mdx [hunk 3]
```diff
@@ -0,0 +1,26 @@
+### Behavior
+
+The **basic-auth** action requires incoming HTTP requests to successfully authenticate
+with HTTP Basic Authentication as per RFC 7235. When a request is received, the action verifies
```
[the remark this answers is not in the document]
Comment (ForgoneReality) [reply traffic-policy/actions/basic-auth/behavior.mdx:4]: ditto: do you want the entire behavior section to be this one-liner, or just the first sentence?

File: traffic-policy/actions/basic-auth/behavior.mdx [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+the request by validating against a known set of `user:password` credentials. If the verification
+is successful, the action allows the request to continue through the action chain and finally
+to your application; if verification fails, the request will be terminated with a
+`401 Unauthorized` response.
+
+#### Additional Restrictions
+
+You can specify only a limited number of `username:password` pairs. The password must be at least
+8 characters and no more than 128 characters. The username can neither be empty nor can it exceed
```
Comment (nijikokun) [inline traffic-policy/actions/basic-auth/behavior.mdx:13]: The third and second sentence say the same thing differently. Could probably remove the third sentence.

Comment (ForgoneReality) [reply traffic-policy/actions/basic-auth/behavior.mdx:13]: ooh nice catch

**Pull request merged**